src/templates/hl7v2/ADT_A01.hbs

```diff
@@ -3,50 +3,99 @@
     "type": "transaction",
     "entry": [
         {{#with (getFirstSegments msg.v2 'PID' 'PD1' 'PV1' 'PV2' 'PR1' 'AVR')}}
-    
-            {{>Resources/Patient.hbs PID=PID PD1=PD1 NK1=NK1 ID=(generateUUID PID-3-1)}},
-            {{>Resources/Encounter.hbs PV1=PV1 PV2=PV2 ID=(generateUUID PV1)}},
-            {{>Resources/Procedure.hbs PR1=PR1 ID=(generateUUID PR1)}},
-            {{>References/Encounter/subject.hbs ID=(generateUUID PV1) REF=(concat 'Patient/' (generateUUID PID-3-1))}}, 
-            {{>References/Procedure/subject.hbs ID=(generateUUID PR1) REF=(concat 'Patient/' (generateUUID PID-3-1))}},	
-            
+
+            {{#if PID-3-1}},
+                {{>Resources/Patient.hbs PID=PID PD1=PD1 NK1=NK1 ID=(generateUUID PID-3-1)}},
+            {{/if}},
+
+            {{#if PV1}},
+ 
+                {{>Resources/Encounter.hbs PV1=PV1 PV2=PV2 ID=(generateUUID PV1)}},
+
+                {{#if PID-3-1}},
+                    {{>References/Encounter/subject.hbs ID=(generateUUID PV1) REF=(concat 'Patient/' (generateUUID PID-3-1))}},
+                {{/if}},
+ 
+            {{/if}},
+
+            {{#if PR1}},
+ 
+                {{>Resources/Procedure.hbs PR1=PR1 ID=(generateUUID PR1)}},
+
+                {{#if PID-3-1}},
+                    {{>References/Procedure/subject.hbs ID=(generateUUID PR1) REF=(concat 'Patient/' (generateUUID PID-3-1))}},
+                {{/if}},
+ 
+            {{/if}},
+
             {{#with (getSegmentLists ../msg.v2 'NK1')}}
                 {{#each NK1 as |NK1Instance|}}
-    
-                    {{>Resources/RelatedPerson.hbs NK1=NK1Instance ID=(generateUUID NK1Instance)}},
-                    {{>References/RelatedPerson/patient.hbs ID=(generateUUID NK1Instance) REF=(concat 'Patient/' (generateUUID ../../PID-3-1))}},
-    
+
+                    {{#if NK1Instance}},
```

Comment:
If NK1Instance is obtained from list (getSegmentLists), can we assume that NK1Instance will never be null or undefined?